src/unix/socket.h

```diff
@@ -67,6 +67,8 @@ struct sock {
     sysreturn (*accept4)(struct sock *sock, struct sockaddr *addr,
             socklen_t *addrlen, int flags);
     sysreturn (*getsockname)(struct sock *sock, struct sockaddr *addr, socklen_t *addrlen);
+    sysreturn (*setsockopt)(struct sock *sock, int level,
```

Comment:
Why not add a getsockopt method too?

Reply:
OK, I'm going to add a getsockopt method too. As of now it is implemented by network sockets only, though.